vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -131,6 +135,8 @@ public Upload() {
                     "Upload cannot be performed without a upload handler set. "
                             + "Please firstly set the upload handler implementation with upload.setUploadHandler()");
         });
+        // event listeners are not needed when no handler is set
+        unregisterUploadHandlerEvents();
```

Comment:
Could we have this NO-OP handler as a specific type so that it wouldn't add the event handlers at all?
Would make it simpler as then there would be no need to unregister in different places.

Reply:
Added a dummy handler and removed registrations.
I think the component doesn't have to clean up these if it switches to a receiver as these won't be invoked anyway.